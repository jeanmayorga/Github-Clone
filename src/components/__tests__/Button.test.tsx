import { render } from '@testing-library/react'
import { Button } from '..'

describe('Render buttons', () => {
  it('should render green button', () => {
    const { container } = render(<Button type="green">Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render outline button', () => {
    const { container } = render(<Button type="outline">Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render disabled button', () => {
    const { container } = render(<Button disabled>Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button with no border left', () => {
    const { container } = render(<Button noBorderLeft>Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button with no border right', () => {
    const { container } = render(<Button noBorderRight>Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render button with no border left and no border right', () => {
    const { container } = render(
      <Button noBorderLeft noBorderRight>
        Button
      </Button>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
