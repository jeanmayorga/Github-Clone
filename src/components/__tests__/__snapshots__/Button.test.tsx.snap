// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render buttons should render button with no border left 1`] = `
<button
  class="rounded-l-none border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
>
  Button
</button>
`;

exports[`Render buttons should render button with no border left and no border right 1`] = `
<button
  class="rounded-l-none rounded-r-none border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
>
  Button
</button>
`;

exports[`Render buttons should render button with no border right 1`] = `
<button
  class="rounded-r-none border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
>
  Button
</button>
`;

exports[`Render buttons should render disabled button 1`] = `
<button
  class="border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
  disabled=""
>
  Button
</button>
`;

exports[`Render buttons should render green button 1`] = `
<button
  class="bg-[#2da44e] hover:bg-[#46954a] active:bg-[#347d39] border-[#cdd9e51a] hover:border-[#cdd9e51a] text-white w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
>
  Button
</button>
`;

exports[`Render buttons should render outline button 1`] = `
<button
  class="disabled:bg-[#f6f8fa] disabled:text-[#0969da80] select-none disabled:cursor-default disabled:border-[#1b1f2426] bg-[#f6f8fa] hover:bg-[#0969da] active:bg-[#0965ce] border-[#1b1f2426] hover:border-[#cdd9e51a] text-[#0969da] hover:text-[#ffffff] active:text-[#ffffff] dark:disabled:bg-[#22272e] dark:disabled:text-[#539bf580] dark:disabled:border-[#cdd9e51a] dark:bg-[#373e47] dark:border-[#cdd9e51a] dark:text-[#539bf5] dark:hover:bg-[#444c56] dark:hover:border-[#cdd9e51a] dark:hover:text-[#539bf5] dark:active:bg-[#0965ce] dark:active:text-white w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
>
  Button
</button>
`;
